### ISSUE TITLE
Startup offers to install Xcode to support "git" if Xcode is not installed

### ISSUE THREAD

**Issue by ronblum** [rstudio/rstudio#2792]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.1.447, 1.2.634
    OS Version      : MacOS 10.13 (High Sierra)
    R Version       : 3.5.0

### Steps to reproduce the problem

Install RStudio on a MacOS that doesn't have Xcode. 

### Describe the problem in detail

Every time RStudio is started, a MacOS dialog appears stating:

> The "git" command requires the command line developer tools. Would you like to install the tools now?

The options are "Get Xcode", "Not Now", and "Install".

### Describe the behavior you expected

I would expect not to get this dialog on startup, since I might not be interested in version control. I would expect it only if I set up version control.

Notes:
- This is the same behavior as trying to run `git` from the command line without Xcode.
- The workaround is to change global options to point to something other than /usr/bin/git:
   1) Tools --> Global Options --> Git/SVN
   2) Git executable --> Browse
   3) Choose a dummy file, e.g. `/usr/bin/at`—anything that doesn't require the developer tools
   4) Open
   5) SVN executable --> Browse
   6) Choose a dummy file, e.g. `/usr/bin/at`—anything that doesn't require the developer tools
   7) Open

This is not a regression—the same behavior occurs in 1.1.447.

**Comment by kevinushey** [comment]
I think this is the expected behavior. Apple packages their own build of `git` as part of the Xcode command line tools and so we prompt the user to install when we discover that `/usr/bin/git` is the version on the `PATH` is the 'stub' that requires command line tool installation.

What happens if you try to execute `/usr/bin/git` from the terminal?

**Comment by jjallaire** [comment]
I believe that this is only supposed to occur when necessary (i.e. not on every startup). Perhaps though the IDE is starting up in a project that has a `.git` directory?

**Comment by ronblum** [comment]
It comes up even if there's no `.git` directory, even after a brand new installation. It also up when I choose `Open Project` and `New Project`. It's the same behavior as if I run `/usr/bin/git` from the terminal. But what I mean to say is, if I have no interest in version control and never chose to do anything with `git`, I still keep getting prompted to install it and Xcode because the Git executable is (correctly) set to `/usr/bin/git`. So I effectively have to tell RStudio and MacOS that I'm not interested in version control by telling it that the `git` executable is something incorrect.

**Comment by JoFAM** [comment]
I have the same issue with a student.
Mac OS 10.14.6
R 4.0.3
RStudio 1.4.1103

git is the one installed with Mac (executable git in /usr/bin), and likely that one insists on having XCode installed. 
Problem persists when disabling the version control interface for RStudio projects in Global Options > Git/SVN

I can't see any reason why RStudio still tries to access git if that one is turned off. Due to security settings on the student's computer, it was impossible to solve the issue by trying to install a different version of git (even though they don't need git, as it's an introductory course).

Any tips or a fix appreciated.

**Comment by kevinushey** [comment]
@ronblum: are you by any chance still able to reproduce this behavior?

**Comment by ronblum** [comment]
Yes, it still occurs. I cheated by renaming `/Library/Developer/CommandLineTools/` to `/Library/Developer/CommandLineTools_BAK`so that `/usr/bin/git` on the command line results in
```
sh-3.2# /usr/bin/git -v
xcode-select: note: no developer tools were found at '/Applications/Xcode.app', requesting install. Choose an option in the dialog to download the command line developer tools.
```

This is in RStudio Desktop 1.4.1106 on MacOS 11.3 beta:
<img width="455" alt="bin_xcode" src="https://user-images.githubusercontent.com/31009952/108234759-1e346680-7113-11eb-951f-d5a9ec861fd7.png">

The workaround is the same as above, changing the Git executable path to a bogus file.

**Comment by kevinushey** [comment]
I see the following instead:

```
$ /usr/bin/git -v
xcrun: error: active developer path ("/Library/Developer/CommandLineTools") does not exist
Use `sudo xcode-select --switch path/to/Xcode.app` to specify the Xcode that you wish to use for command line developer tools, or use `xcode-select --install` to install the standalone command line developer tools.
```

**Comment by kevinushey** [comment]
Took some time to dive in; we indeed do try to initialize a bunch of git behavior even when a project hasn't explicitly been configured to use `git`. This happens in a couple spots:

https://github.com/rstudio/rstudio/blob/b6a0f8fd4e841ee8e1c856356983180be4fa017e/src/cpp/session/modules/SessionGit.cpp#L3324

https://github.com/rstudio/rstudio/blob/b6a0f8fd4e841ee8e1c856356983180be4fa017e/src/cpp/session/modules/SessionVCS.cpp#L259-L270

I'm a little wary of touching this code, though -- I don't think we can fix this for 1.4-JR.
